### ISSUE TITLE
Sparse Arrays

### ISSUE THREAD

**Issue by lukecampbell** [ooici/coverage-model#175]
I can successfully set data as an array but a ValueError is raised when attempting to call get_parameter_values

```
Traceback (most recent call last):
  File "/Users/luke/Documents/Dev/code/coi2/extern/coverage-model/coverage_model/test/test_postgres_storage.py", line 561, in test_sparse_arrays
    cov.get_parameter_values().get_data()
  File "/Users/luke/Documents/Dev/code/coi2/extern/coverage-model/coverage_model/coverage.py", line 401, in get_parameter_values
    vals = self._persistence_layer.read_parameters(param_names, time_segment, time, sort_parameter, stride_length=stride_length, fill_empty_params=fill_empty_params)
  File "/Users/luke/Documents/Dev/code/coi2/extern/coverage-model/coverage_model/storage/parameter_persisted_storage.py", line 268, in read_parameters
    np_dict, function_params, rec_arr = self.get_data_products(params, time_range, time, sort_parameter, stride_length=stride_length, create_record_array=True, fill_empty_params=fill_empty_params)
  File "/Users/luke/Documents/Dev/code/coi2/extern/coverage-model/coverage_model/storage/parameter_persisted_storage.py", line 280, in get_data_products
    np_dict = self._create_parameter_dictionary_of_numpy_arrays(numpy_params, function_params, stride_length=stride_length, params=dict_params)
  File "/Users/luke/Documents/Dev/code/coi2/extern/coverage-model/coverage_model/storage/parameter_persisted_storage.py", line 419, in _create_parameter_dictionary_of_numpy_arrays
    fill_value=self.value_list[param_name].fill_value)
  File "/Users/luke/Documents/Dev/code/coi2/extern/coverage-model/coverage_model/parameter_data.py", line 155, in merge_data_as_numpy_array
    arr = obj_dict[key].get_data_as_numpy_array(alignment_array, fill_value=fill_value, arr=arr)
  File "/Users/luke/Documents/Dev/code/coi2/extern/coverage-model/coverage_model/parameter_data.py", line 182, in get_data_as_numpy_array
    arr[alignment_array >= self.start] = self._data
ValueError: NumPy boolean array indexing assignment cannot assign 2 input values to the 20 output values where the mask is true
```

**Comment by caseybryant** [comment]
I believe this was resolved with commit 459de8126e4972cf5900306670fc9ec9e1b4a039.  Can you verify?

**Comment by lukecampbell** [comment]
Yeah this one is done.
